src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/workflow.html.twig

```diff
@@ -23,6 +145,54 @@
                 flowchart: { useMaxWidth: false },
                 securityLevel: 'loose',
             });
+
+            {% for name, data in collector.workflows %}
+                window.showNodeDetails{{ collector.hash(name) }} = function (node) {
+                    const map = {{ data.listeners|json_encode|raw }};
+                    showNodeDetails(node, map);
+                };
+            {% endfor %}
+
+            const showNodeDetails = function (node, map) {
+                const dialog = document.getElementById('detailsDialog');
+                dialog.querySelector('tbody').innerHTML = '';
+                for (const [eventName, listeners] of Object.entries(map[node])) {
+                    listeners.forEach(listener => {
+                        const row = document.createElement('tr');
+                        let listenerDetails = `<code>${listener.title}</code>`;
+                        if (listener.file) {
+                            listenerDetails = `<a href="${listener.file}">${listenerDetails}</a>`;
+                        }
+                        if (typeof listener.guardExpressions === 'object') {
+                            console.log('XXX"');
+                            listenerDetails = `${listenerDetails} (guard expressions: <code>${listener.guardExpressions.join('</code>, <code>')})</code>`;
+                        }
+                        row.innerHTML = `
+                            <td><code>${eventName}</code></td>
+                            <td>${listenerDetails}</td>
```

Comment:
this misses HTML escaping. This is a no-go.

Reply:
if you don't want to care about escaping manually, the solution is to create DOM nodes instead of creating HTML (and then, you can assign `textContent` or create text nodes)

Reply:
maybe a simpler way would be to create the whole dialog in Twig, with different dialogs per workflow.

Reply:
I did that initially, but the HTML will be SO huge! 
N workflow * M transition * O place === the number of modal :(

Reply:
ah indeed. then use proper DOM building.

Reply:
Fixed ✅